app/models/document.rb

```diff
@@ -2,11 +2,12 @@ class Document
   include Mongoid::Document
   include Mongoid::Timestamps
 
+  mount_uploader :source, DocumentUploader
 
   ACCESS_RIGHTS = %w(public pii_restricted)
 
   # Enable polymorphic associations
-  embedded_in :documentable, polymorphic: true
+  #embedded_in :documentable, polymorphic: true
```

Comment:
Due to relationship errors. once we finalize the model to save document I will revert back that comment.